src/components/auth-forms/LoginForm.jsx

```diff
@@ -1,12 +1,58 @@
-import { useState } from "react";
 import "./auth-forms.css";
+import { useState, useEffect } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { Loader } from "../index";
+import { Toast } from "../../utils";
+import { useAuth } from "../../context";
 export const LoginForm = () => {
+  const navigate = useNavigate();
   const [showPwd, setShowPwd] = useState(false);
+  const [loginCredentials, setLoginCredentials] = useState({
+    email: "",
+    password: "",
+  });
+  const { userAuthDispatch, loginHandler, serverResponse, isLoaderLoading } =
+    useAuth();
+  const submitHandler = (e) => {
+    e.preventDefault();
+    loginHandler(loginCredentials);
+    setLoginCredentials((prev) => ({ ...prev, email: "", password: "" }));
+  };
+  useEffect(() => {
+    if (serverResponse) {
+      const { data, status } = serverResponse;
+      if (status === 200) {
+        localStorage.setItem("token", data.encodedToken);
+        userAuthDispatch({
+          type: "LOGIN",
+          payload: {
+            encodedToken: data.encodedToken,
+            user: { ...data.foundUser },
+          },
+        });
+
+        navigate("/");
+        Toast({ type: "success", msg: "Login Successful!" });
```

Comment:
Also, handle the serverError as well, so that if the user posts incorrect credentials or some other error occurs, it can be caught

Reply:
Sure I will do that.